### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by amorehead**
#### Reference Issues/PRs
I noticed a TODO statement that I could readily address.

#### What does this implement/fix? Explain your changes
I added a way of checking that DSSP is executable before trying to use it.

#### What testing did you do to verify the changes in this PR?
I made sure the code runs locally.

### CODE REVIEW

**Comment by a-r-j** [comment]
Thanks @amorehead ! Appreciate the PR :) Commented on a few nits. 

Would you also mind updating the [changelog](https://github.com/a-r-j/graphein/blob/master/CHANGELOG.md) and add yourself to the [contributors](https://github.com/a-r-j/graphein/blob/master/CONTRIBUTORS.md) if you wish?

**Comment by amorehead** [comment]
@a-r-j,

No problem! Happy to contribute where I can. I've updated the `CHANGELOG.md` and `CONTRIBUTORS.md` files as you mentioned (https://github.com/a-r-j/graphein/pull/154/commits/847187626edb7552de0256d88901ab683e2776e6), however, I cannot see your latest comments on my proposed changes. Any ideas why that may be?

**Review by a-r-j — CHANGES_REQUESTED** [review]

**Comment by a-r-j** [comment]
> @a-r-j,
> 
> No problem! Happy to contribute where I can. I've updated the `CHANGELOG.md` and `CONTRIBUTORS.md` files as you mentioned ([8471876](https://github.com/a-r-j/graphein/commit/847187626edb7552de0256d88901ab683e2776e6) and [441f94e](https://github.com/a-r-j/graphein/commit/441f94e91f2df95e6634973e0fb3514bde457759)), however, I cannot see your latest comments on my proposed changes. Any ideas why that may be?

Apologies, forgot to submit the review :)

File: graphein/protein/features/nodes/dssp.py [hunk 1]
```diff
@@ -109,7 +109,8 @@ def add_dssp_df(G: nx.Graph, dssp_config: Optional[DSSPConfig]) -> nx.Graph:
     config = G.graph["config"]
     pdb_id = G.graph["pdb_id"]
 
-    # TODO - Check for DSSP installation
+    # Ensure that DSSP is on PATH and is marked as an executable.
```
[the remark this answers is not in the document]
Comment (amorehead) [reply graphein/protein/features/nodes/dssp.py:115]: Sure thing. These changes have been addressed with in [this new commit](https://github.com/amorehead/graphein/commit/726890925f7ed50d2d797b726ee8cc556c6e0bd5).

File: graphein/protein/utils.py [hunk 2]
```diff
@@ -299,3 +299,15 @@ def save_rgroup_df_to_pdb(g: nx.Graph, path: str, gz: bool = False):
     ppd.df["ATOM"] = g.graph["rgroup_df"]
     ppd.to_pdb(path=path, records=None, gz=gz, append_newline=True)
     log.info(f"Successfully saved rgroup data to {path}")
+
+
+def is_tool(name: str):
```
[the remark this answers is not in the document]
Comment (amorehead) [reply graphein/protein/utils.py]: Sure thing. These changes have been addressed in [this new commit](https://github.com/amorehead/graphein/commit/726890925f7ed50d2d797b726ee8cc556c6e0bd5).

File: graphein/protein/utils.py [hunk 3]
```diff
@@ -299,3 +299,15 @@ def save_rgroup_df_to_pdb(g: nx.Graph, path: str, gz: bool = False):
...
+    """Checks whether ``name`` is on PATH and is marked as an executable.
+    
+    Source: https://stackoverflow.com/questions/11210104/check-if-a-program-exists-from-a-python-script
+
+    :param name: Name of program to check for execution ability.
+    :type name: str
```
[the remark this answers is not in the document]
Comment (amorehead) [reply graphein/protein/utils.py:311]: Sure thing. These changes have been addressed in [this new commit](https://github.com/amorehead/graphein/commit/726890925f7ed50d2d797b726ee8cc556c6e0bd5).

File: graphein/protein/utils.py [hunk 4]
```diff
@@ -299,3 +299,15 @@ def save_rgroup_df_to_pdb(g: nx.Graph, path: str, gz: bool = False):
...
+    """
+    from shutil import which
```
[the remark this answers is not in the document]
Comment (amorehead) [reply graphein/protein/utils.py]: Sure thing. These changes have been addressed in [this new commit](https://github.com/amorehead/graphein/commit/726890925f7ed50d2d797b726ee8cc556c6e0bd5).

**Pull request merged**